src/components/vanilla/charts/DynamicAxisBar/index.tsx

```diff
@@ -0,0 +1,52 @@
+import { useEmbeddableState } from '@embeddable.com/react';
+import React, { useState, useEffect, useRef } from 'react';
+import Container from '../../Container';
+import Dropdown from '../../controls/Dropdown';
+import BarChart from '../BarChart/components/BarChart';
+
+export default (props: Props) => {
+  const { results, title } = props;
+
+  const [value, setValue] = useState(props.xAxis?.title);
+  const [_, setDimension] = useEmbeddableState({ dimension: null });
+
+  const xAxisOptions = props.xAxisOptions?.find((item) => props.xAxis.title == item.title) 
+    ? props.xAxisOptions 
+    : [...props.xAxisOptions, props.xAxis ]
+
+
+
+  const handleChange = (newValue) => {
+    setValue(newValue);
+    const selectedDimension = xAxisOptions.find(item => newValue === item.title);
+    setDimension({ dimension: selectedDimension });
+  }
+
+  const xAxis = xAxisOptions.find(item => value === item.title);
+  const updatedProps = {...props, xAxis: xAxis };
+
+  return (
+    <Container
+      {...props}
+      className="overflow-y-hidden">
+      <div className="flex h-[60px] w-full">
+          <div>
+            <select className="relative px-3 rounded-xl w-full min-w-[50px] h-10 border border-[#DADCE1] flex items-center" value={value} onChange={(e) => handleChange(e.target.value)} style={{ position: 'relative', height: '40px', width: 'auto'}}>
```

Comment:
Is that style definition necessary? position is duplicated with class `relative`, height could be replaced by class `h-[50px]` a width with `w-auto`

Reply:
@kuka-radovan good point, but I added this due to a really strange bug with stencil + lastpass (that our team know about). Without this, the layout doesn't work properly when you click for the first time.

Reply:
But I will remove this anyway, as it's such an edge case.

Reply:
I have no problem with that if it is intentional. I don't know the circumstances.